docs/flow/config-language/expressions/operators.md

```diff
@@ -6,3 +6,96 @@ weight: 300
 ---
 
 # Operators
+River uses a set of operators that most should be familiar with. All operations
+follow the standard [PEMDAS](https://en.wikipedia.org/wiki/Order_of_operations)
+rule for operator precedence.
+
+## Arithmetic operators
+```
++    for addition: 3+5
+-    for subtraction: 10-3
+*    for multiplication: 13*7
+/    for division: 15/3
+%    for remainder: 7%2
+^    for exponentiation: 2^8
+```
+
+The `+` operator can also be used for string concatenation.
+
+## Comparison operators
+```
+==    equal
+!=    not equal
+<     less
+<=    less or equal
+>     greater
+>=    greater or equal
+```
+
+The equality operators `==` and `!=` can be applied to any operands.
+
+On the other hand, for the ordering operators the two operands must both be
+_orderable_ and of the same type. These terms and the result of the comparisons
+are defined as follows:
+
+* Boolean values are comparable. Two boolean values are equal if they are
+  either both true or both false.
+* Numerical (integer and floating-point) values are comparable and ordered, in
+  the usual way.
+* String values are comparable and ordered, lexically byte-wise.
+* Objects are comparable. They are are equal if all their fields are equal.
+* Array values are comparable. They are equal if their corresponding elements
+  are equal.
+
+## Logical operators
+```
+&&    conditional AND
+||    conditional OR 
+!     NOT            
+```
+
+Logical operators apply to boolean values and yield a boolean result.
+
+## Assignment operator
+River uses `=` as its assignment operator.
+
+An assignment statement may only assign a single value to a single attribute.
+In assignments, each value must be _assignable_ to the attribute to which it is
+assigned.
+
+* The `null` value can be assigned to any attribute.
+* Numerical, string, boolean, array and object types are assignable to
+  attributes of the corresponding type.
```

Comment:
I think that _functions_ are also assignable, right?

Reply:
Yeah, functions are first-class values here. Capsules are also assignable. 

There are also special exceptions where some types can be [converted](https://github.com/grafana/agent/blob/cbbfabb8b7844d702409dc62dcc32fd330973ec5/pkg/river/internal/value/value.go#L463):

- Numbers can always be converted to strings 
- Strings can be converted to numbers if the string represents a number (otherwise evaluation will fail)

Reply:
Okay, I'll add some bulletpoints to the assignment rules!